File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        ans = []
        hashMap = {}
        for i in range(0, len(groupSizes)):
            if groupSizes[i] not in hashMap:
                hashMap[groupSizes[i]] = []
            hashMap[groupSizes[i]].append(i)
            if len(hashMap[groupSizes[i]]) == groupSizes[i]:
                ans.append(hashMap[groupSizes[i]])
                hashMap[groupSizes[i]] = []
        return ans

File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.java
class Solution {
    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        List<List<Integer>> ans = new ArrayList<>();
        Map<Integer, List<Integer>> hash = new HashMap<>();
        for(int i=0; i<groupSizes.length; i++){
            if(!hash.containsKey(groupSizes[i])){
                hash.put(groupSizes[i], new ArrayList<>());
            }
            List<Integer> group = hash.get(groupSizes[i]);
            group.add(i);
            if(group.size() == groupSizes[i]){
                ans.add(group);
                hash.remove(groupSizes[i]);
            }
        }
        return ans;
    }
}